packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/upgrade/upgrade.controller.js

```diff
@@ -0,0 +1,170 @@
+import 'moment';
+
+import {
+  COMPOSITE_UPGRADE,
+  MIN_INTERVENTION_GAP,
+  UPGRADE_MODE,
+} from './upgrade.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($http, $translate, $window, Alerter, coreURLBuilder) {
+    this.$http = $http;
+    this.$window = $window;
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.coreURLBuilder = coreURLBuilder;
+
+    this.UPGRADE_MODE = UPGRADE_MODE;
+  }
+
+  $onInit() {
+    this.interventionData = {
+      backupDone: false,
+      selectedDate: null,
+      selectedSlot: null,
+      timeSlots: [],
+    };
+    this.upgradeStarted = this.upgradeMode !== this.UPGRADE_MODE.ORDER;
+    this.upgradeScheduled = this.upgradeMode === this.UPGRADE_MODE.NONE;
+    if (this.upgradeStarted) {
+      this.currentStep = 1;
+    }
+    if (this.upgradeScheduled) {
+      this.currentStep = 2;
+    }
+    if (this.upgradeStarted) {
+      this.upgradeType =
+        this.upgradeTask.components.length > 1
+          ? COMPOSITE_UPGRADE
+          : this.upgradeTask.components[0];
+    }
+  }
+
+  onDateChange([selectedDate]) {
+    this.loadingTimeSlots = true;
+    let periodStart = moment(selectedDate).subtract(1, 'days');
+    periodStart = (periodStart.isBefore(this.interventionData.minDate)
+      ? moment(this.interventionData.minDate)
+      : periodStart
+    ).format('YYYY-MM-DD');
```

Comment:
if you are displaying this on UI, consider formating based on local

Reply:
This data is just being passed to the APIs